fortitude/src/rules/correctness/exit_labels.rs

```diff
@@ -149,6 +149,100 @@ impl AstRule for ExitOrCycleInUnlabelledLoop {
     }
 }
 
+/// ## What does it do?
+/// Check for missing block labels on `end` statements
+///
+/// ## Why is this bad?
+/// This is a compile error!
+#[derive(ViolationMetadata)]
+pub(crate) struct MissingBlockLabel {
+    start_name: String,
+    end_name: String,
+    label: String,
+}
+
+impl AlwaysFixableViolation for MissingBlockLabel {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        let Self {
+            start_name,
+            end_name,
+            ..
+        } = self;
+        format!("'{end_name}' statement in named '{start_name}' block missing label")
+    }
+
+    fn fix_title(&self) -> String {
+        let Self { label, .. } = self;
+        format!("Add label '{label}'")
+    }
+}
+
+impl AstRule for MissingBlockLabel {
+    fn check<'a>(
+        _settings: &Settings,
+        node: &'a Node,
+        src: &'a SourceFile,
+    ) -> Option<Vec<Diagnostic>> {
+        let src = src.source_text();
+        // Skip unlabelled loops
+        let label = node
+            .child_with_name("block_label_start_expression")?
+            .to_text(src)?
+            .trim_end_matches(':');
+
+        let end = match node.kind() {
+            "select_case_statement" | "select_type_statement" | "select_rank_statement" => {
```

Comment:
I missed `select_rank_statement` in `keywords-missing-space`, I'll have to raise an issue for that

Reply:
Actually, it seems `selectrank` isn't valid, but tree-sitter parses it as if it were.